Frends.Community.SQL/Frends.Community.SQL.cs

```diff
@@ -226,6 +227,90 @@ private static Encoding GetEncoding(FileEncoding optionsFileEncoding, bool optio
                 default:
                     throw new ArgumentOutOfRangeException();
             }
+        }
+
+
+        /// <summary>
+        /// Bulk insert DataTable to a SQL table.
+        /// </summary>
+        /// <param name="input">Input parameters</param>
+        /// <param name="options">Optional parameters with default values</param>
+        /// <param name="cancellationToken">Cancellation token</param>
+        /// <returns>Copied row count</returns>
+        public static async Task<int> BulkInsertDataTable([PropertyTab] BulkInsertInput input, [PropertyTab] BulkInsertOptions options, CancellationToken cancellationToken)
+        {
+            using (var connection = new SqlConnection(input.ConnectionString))
+            {
+                await connection.OpenAsync(cancellationToken).ConfigureAwait(false);
+                cancellationToken.ThrowIfCancellationRequested();

```

Comment:
No need to check cancellation here. Cancellation should be manually checked only in iterations and loops which might take a while.